File: 3157-find-the-level-of-tree-with-minimum-sum/3157-find-the-level-of-tree-with-minimum-sum.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
typedef pair<int, int> pii;
#define MAX numeric_limits<ll>::max()
typedef long long ll;

class Solution {
public:
    int minimumLevel(TreeNode* root) {
        int ans = 0;
        ll sum = MAX;
        queue<TreeNode*> q;
        q.push(root);
        int level = 1;
        while (!q.empty()) {
            int n = q.size();
            ll currSum = 0;

            for (int i = 0; i < n; ++i) {
                TreeNode* node = q.front();
                q.pop();

                currSum += node->val;

                if (node->left)
                    q.push(node->left);
                if (node->right)
                    q.push(node->right);
            }

            if (currSum < sum) {
                sum = currSum;
                ans = level;
            }
            level++;
        }

        return ans;
    }
};

File: 3157-find-the-level-of-tree-with-minimum-sum/3157-find-the-level-of-tree-with-minimum-sum.ts
/**
 * Definition for a binary tree node.
 * class TreeNode {
 *     val: number
 *     left: TreeNode | null
 *     right: TreeNode | null
 *     constructor(val?: number, left?: TreeNode | null, right?: TreeNode | null) {
 *         this.val = (val===undefined ? 0 : val)
 *         this.left = (left===undefined ? null : left)
 *         this.right = (right===undefined ? null : right)
 *     }
 * }
 */

function minimumLevel(root: TreeNode | null): number {
    const q: [TreeNode] = [root];
    let ans: number = 0, level: number = 1;
    let sum: number = Infinity;

    while (q.length > 0) {
        let n: number = q.length;
        let cur: number = 0;

        for (let i: number = 0; i < n; i++) {
            let node: TreeNode = q.shift();
            cur += node.val;

            if (node.left)
                q.push(node.left);
            if (node.right)
                q.push(node.right);
        }

        if (cur < sum) {
            sum = cur;
            ans = level;
        }
        level++;
    }

    return ans;
};